test/common/settings.cpp

```diff
@@ -0,0 +1,118 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+//
+// Tests for include/measurement_kit/common/settings.hpp
+//
+
+#define CATCH_CONFIG_MAIN
+#include "src/ext/Catch/single_include/catch.hpp"
+
+#include <measurement_kit/common/settings.hpp>
+#include <string>
+
+using namespace mk;
+
+TEST_CASE("The settings class convert string to int") {
+    Settings settings;
+    settings["key"] = "6";
+    int value = settings["key"].as<int>();
+    REQUIRE(value == 6);
+}
+
+TEST_CASE("The settings class throw an exception when the conversions fails, "
+          "due to an empty string") {
+    Settings settings;
+    settings["key"] = "";
+    REQUIRE_THROWS_AS(settings["key"].as<int>(), ValueError);
+}
+
+TEST_CASE("The settings class throw an exception when the conversion is not "
+          "complete") {
+    Settings settings;
+    settings["key"] = "4 5 6";
+    REQUIRE_THROWS_AS(settings["key"].as<int>(), ValueError);
+}
+
+TEST_CASE("The settings class convert bool to int") {
+    Settings settings;
+    settings["key"] = true;
+    REQUIRE(settings["key"].as<int>() == 1);
+}
+
+TEST_CASE("The settings class convert bool to string") {
+    Settings settings;
+    settings["key"] = true;
+    REQUIRE(settings["key"].as<std::string>() == "1");
+}
+
+TEST_CASE("The settings class convert bool to double") {
+    Settings settings;
+    settings["key"] = true;
+    REQUIRE(settings["key"].as<double>() == 1);
+}
+
+TEST_CASE(
+    "The settings class throws an exception while converting double to int") {
+    Settings settings;
+    settings["key"] = 6.5;
+    REQUIRE_THROWS_AS(settings["key"].as<int>(), ValueError);
+}
+
+TEST_CASE(
+    "The settings class throws an exception when converting double to bool") {
+    Settings settings;
+    settings["key"] = 6.5;
+    REQUIRE_THROWS_AS(settings["key"].as<bool>(), ValueError);
+}
+
+TEST_CASE("The settings class converts double to string") {
+    Settings settings;
+    settings["key"] = 6.5;
+    REQUIRE(settings["key"].as<std::string>() == "6.5");
+}
+
+TEST_CASE("The settings class converts int to double") {
+    Settings settings;
+    settings["key"] = 6;
+    REQUIRE(settings["key"].as<double>() == 6.0);
+}
+
+TEST_CASE("The settings class converts int to string") {
+    Settings settings;
+    settings["key"] = 6;
+    REQUIRE(settings["key"].as<std::string>() == "6");
+}
+
+TEST_CASE(
+    "The settings class throws an exception when converting int (10) to bool") {
+    Settings settings;
+    settings["key"] = 10;
+    REQUIRE_THROWS_AS(settings["key"].as<bool>(), ValueError);
+}
```

Comment:
This is suprising: I would have expected the conversion to succeed and return `true`.